src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketDeploymentClientImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeployment;
+import okhttp3.Headers;
+import okhttp3.HttpUrl;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.lang3.StringUtils.stripToNull;
+
+public class BitbucketDeploymentClientImpl implements BitbucketDeploymentClient {
+
+    private final BitbucketRequestExecutor bitbucketRequestExecutor;
+    private final String projectKey;
+    private final String repoSlug;
+    private final String revisionSha;
+
+    public BitbucketDeploymentClientImpl(BitbucketRequestExecutor bitbucketRequestExecutor, String projectKey,
+                                         String repoSlug, String revisionSha) {
+        this.bitbucketRequestExecutor = requireNonNull(bitbucketRequestExecutor, "bitbucketRequestExecutor");
+        this.projectKey = requireNonNull(stripToNull(projectKey), "projectKey");
+        this.repoSlug = requireNonNull(stripToNull(repoSlug), "repoSlug");
+        this.revisionSha = requireNonNull(stripToNull(revisionSha), "revisionSha");
+    }
+
+    @Override
+    public void post(BitbucketDeployment deployment) {
+        HttpUrl url = bitbucketRequestExecutor.getBaseUrl().newBuilder()
```

Comment:
do we need to check the CD capabilities before making this request? Otherwise `getCDCapabilities()` is not used currently

Reply:
Currently, it's not used but I think that the job of this client should just be to send the data off to Bitbucket. Since this class doesn't know what should happen when Bitbucket doesn't support deployments (should it be an error, warning or just silently ignored?) I think it should be the caller's job to call the capabilities endpoint first and decide what to do based on that.